### PR TITLE
[Product Block Editor]: Introduce ButtonWithDropdownMenuProps component

### LINKED ISSUE #43104: Introduce a Button + Dropdown component
We are looking for a component that looks like the one shown in the figure below:

* A primary button (`Add to store` in this example)
* A button that triggers the dropdown menu

Required in https://github.com/woocommerce/woocommerce/issues/43046, https://github.com/woocommerce/woocommerce/issues/43047, https://github.com/woocommerce/woocommerce/issues/40376, ...

### PR DESCRIPTION
**Pull request by retrofox**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR introduces the `<ButtonWithDropdownMenu />` component, which will be required to implement https://github.com/woocommerce/woocommerce/issues/43046, https://github.com/woocommerce/woocommerce/issues/43047, https://github.com/woocommerce/woocommerce/issues/40376, ...

Closes https://github.com/woocommerce/woocommerce/issues/43104

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Let's use Storybook

1. Update, just in case:

```cli
pnpm install
````

2. Run the Storybook app

```cli
pnpm --filter=storybook run storybook
```

3. Go to the ShippingBags story: 

`http://localhost:6007/?path=/story/product-editor-components-buttonwithdropdownmenu--default`

4. Confirm you see the story
5. Play with the component properties

https://github.com/woocommerce/woocommerce/assets/77539/7a1472a4-6148-44cb-9cfa-9e6238ad806f

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/components/button-with-dropdown-menu/index.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	defaultOpen = false,
+	popoverProps: {
+		placement = 'bottom-end',
+		position = 'bottom left left',
+		offset = 0,
+	} = {
+		placement: 'bottom-end',
+		position: 'bottom left left',
+		offset: 0,
+	},
```
Comment (octaedro) [inline packages/js/product-editor/src/components/button-with-dropdown-menu/index.tsx:30]: I wonder if it's worth adding a `className` prop to provide more flexibility. What do you think about that?

**Review by octaedro — APPROVED** [review]
Nice job @retrofox! This is testing well on my end and the code looks good!
I just left a small question, but I'll pre-approve this PR

File: packages/js/product-editor/src/components/button-with-dropdown-menu/index.tsx (hunk 1 above)
Comment (retrofox) [reply packages/js/product-editor/src/components/button-with-dropdown-menu/index.tsx:30]: SGTM! Let me do it in a follow-up/ 🙇‍♂️

Comment (retrofox) [reply packages/js/product-editor/src/components/button-with-dropdown-menu/index.tsx:30]: https://github.com/woocommerce/woocommerce/pull/43136

**Pull request merged**